### PR TITLE
[FIVE-166 - Front] Refactorizar el código de los controladores y servicios de artefactos

### PR DESCRIPTION
**Pull request by gperelstein**
-Se modificaron los componentes Routes. y authService para hacer que el componente que muestra los artifacts (ArtifactsDetails) tenga una ruta privada
-Se modificó el ArtifactsDetails para que al transformar el projectId de string a number se haga uso de la función parseInt

### CODE REVIEW

File: FiveRockingFingers/FRF.Web/ClientApp/src/components/auth/authService.tsx [hunk 1]
```diff
@@ -1,9 +1,10 @@
 import React from "react";
-import { Route, Redirect } from "react-router-dom";
+import { Route, Redirect, match, RouteComponentProps } from "react-router-dom";
 import { useUserContext } from "./contextLib";
 
+
 const PrivateRoute: React.FC<{
```
Comment (matiasbeckerle) [inline FiveRockingFingers/FRF.Web/ClientApp/src/components/auth/authService.tsx:6]: Curious: what's a private route in react's world?

File: FiveRockingFingers/FRF.Web/ClientApp/src/router/Routes.tsx [hunk 2]
```diff
@@ -10,13 +10,13 @@ import ManageProjects from '../components/ManageProjects';
...
-	    <Redirect exact from="/" to="/home" />
+        <Redirect exact from="/" to="/home" />
+        <Route exact path='/' component={Home} />
         <Route exact path="/login" component={Login} />
         <Route exact path="/signup" component={Signup} />
         <PrivateRoute exact path="/home" component={Home} />
         <PrivateRoute exact path='/administrarProyectos' component={ManageProjects} />
-        <Route exact path='/projects/:idProject/artifacts/' component={ArtifactsDetails} />
-        <Route exact path='/' component={Home} />
+        <PrivateRoute exact path='/projects/:idProject/artifacts/' component={ArtifactsDetails} />
```
Comment (matiasbeckerle) [inline FiveRockingFingers/FRF.Web/ClientApp/src/router/Routes.tsx:19]: `:projectId` maybe?

**Review by matiasbeckerle — APPROVED** [review]

File: FiveRockingFingers/FRF.Web/ClientApp/src/components/auth/authService.tsx (hunk 1 above)
Comment (gonzalorom) [reply FiveRockingFingers/FRF.Web/ClientApp/src/components/auth/authService.tsx:6]: In this app, `PrivateRoute` is a functional component to allow an authorized user to go to the requested URL. On another hand, you have `Route` that is the opposite, any user is allowed.

Comment (matiasbeckerle) [reply FiveRockingFingers/FRF.Web/ClientApp/src/components/auth/authService.tsx:6]: Ah private in that sense. Gotcha, thanks for explaining!

File: FiveRockingFingers/FRF.Web/ClientApp/src/router/Routes.tsx (hunk 2 above)
Comment (gperelstein) [reply FiveRockingFingers/FRF.Web/ClientApp/src/router/Routes.tsx]: Done!

**Review by gonzalorom — APPROVED** [review]

**Comment by matiasbeckerle** [comment]
Careful before merging: GH indicates this one has some conflicting changes.

**Comment by gperelstein** [comment]
Yes, this branch has conflicts that I knew I was going to have when the branch FIVE-115 was merged into Release_01. I already solved it.

**Pull request merged**